src/services/conversionRates/getConversionRatesService.js

```diff
@@ -24,6 +24,56 @@ const _getRatesDb = async (app, timestamp, symbol = 'ETH') => {
   return { timestamp, rates: {} };
 };
 
+/**
+ * Get the rates from the coingecko API
+ *
+ * @throws Error if fetching the rates from cryptocompare API failed
+ *
+ * @param {Number} timestamp   Timestamp for which the value should be retrieved
+ * @param {Array}  ratestToGet Rates that are missing in the DB and should be retrieved
+ * @param {Array}  stableCoins coins whose value equal one usd
+ *
+ * @return {Object} Rates object in format { EUR: 241, USD: 123 }
+ */
+const _getRatesCoinGecko = async (rateSymbol, timestampMS, coingeckoId) => {
+  const today = new Date(new Date().setUTCDate(new Date().getUTCDate()));
+  const todayUTC = today.setUTCHours(0, 0, 0, 0);
+  const todayRounded = Math.round(todayUTC / 1000);
+  const differenceDays = (todayRounded - timestampMS) / 86400;
+  const daysMax = Math.round(differenceDays);
+  const testRep = JSON.parse(
+    await rp(
+      `https://api.coingecko.com/api/v3/coins/${coingeckoId}/market_chart?vs_currency=${rateSymbol}&days=${daysMax}`,
```

Comment:
Let's use `/coins/{id}/market_chart/range` api for efficiency. Send request for some finite range before and after the requested timestamp, 10 minutes maybe enough.

Reply:
Didn't understood this

Reply:
Oh i understand this one, i will check it out!